mtt-8638-prtriggers-per-project

harshala-rajesh opened the pull request:
### Description

Adjust Samples tests to be more efficient
Creating separate PR Triggers for each project in multiplayer.samples.bitesize , so only relevant test to the project is run when a change is made instead of running tests for all the projects. Details in the ticket

### Issue Number(s)

(https://jira.unity3d.com/browse/MTT-8638)

### Contribution checklist
 - [x] Tests have been added for the project and/or any internal package
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=185) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=185) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=185) it.</sub>

RikuTheFuffs commented:
@harshala-rajesh 
1. it seems that tests are failing, could you investigate why?
2. The branch of this PR should start from `develop`, and it should target `develop`, not `main`
3. Please sign the CLA agreement on this github page (see https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/pull/185#issuecomment-2166516939 ) otherwise the PR won't be able to go through

RikuTheFuffs commented on .yamato/2DSpaceShooter_triggers.yml:
```diff
@@ -0,0 +1,48 @@
+pull_request_trigger:
+  name: 2DSpaceShooter Pull Request Trigger (main, develop, & sample branches)
+  dependencies:
+{% for project in projects -%}
+{% if project.name =="2dspaceshooter" -%}
+{% for editor in project.test_editors -%}
+{% for platform in test_platforms -%}
```
there's a lot of repetition in this part across samples triggers, where the only difference is `{% if project.name =="<project name>" -%}`
is there any way we can avoid repeating all this code several times? Maybe by adding a cname check in the "paths" section instead? Or it is impossible due to Yamato's nature?

harshala-rajesh replied:
Yes, it is going through all projects and running further steps only if it is a specific project, that is reason I added if condition at line 9 , maybe there is a way , let me check, but for now this seems to be working for our purposes, it only starts jobs specific to the project.

harshala-rajesh commented:
Adding Conversation I had with Paolo on slack
Harshala Rajesh
Paolo, I was wondering if we could merge my [PR](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/pull/185) on develop. This PR essentially removes single trigger which was running all jobs for any PR and separates it into 5 different Triggers for 5 folders and each job should be triggered only when the content of specific folder changes. Based on consultation with devs-yamato, I need to merge this PR  and then test by changing content in each folder and have a separate PR for them. If develop is not being actively used by the team , could my PR be merged and I will test say for 2-3 days to confirm that is working as desired. I am at offsite till June 21, Friday. So if we can merge by Monday, I will have it tested out by Wednsesday .

Paolo Abela
Hi Harshala, unfortunately develop is actively used by the team, and it's where we release things from, so only verified changes should go there.
If my understanding is correct, You should be able to test this PR by:
creating a branch from the one of the PR
opening a pull request that targets your current PR branch.
make a commit for each sample to simulate a change
Also, could you paste these comments on the PR, so other team members can be aware and join the discussion? (edited) 

Harshala Rajesh
Ok will do

Paolo Abela
For point (3), you could simply add a file under each sample's folder to see if it triggers the tests of that sample only (edited) 


Harshala Rajesh
I was trying to make a commit with change to each folder in the current PR , but that did not work as the trigger looks at change for entire PR  and thus trigger all jobs


Paolo Abela
i think you'll need to test such changes in another branch and PR, because that one should react to the Yamato changes you made in this current branch.
But please, let's move this conversation to the public PR. The rest of the team needs visibility over this.

RikuTheFuffs reviewed (changes requested):
@harshala-rajesh could you please remove all test files?

harshala-rajesh commented:
oops.. missed it... done ...

RikuTheFuffs reviewed (changes requested):
@harshala-rajesh it seems that the `[Yamato] Pull Request Trigger` part of the CI is not running, could it be because it expects a file called `tests.yml` , which was renamed in favor of `shared_triggers.yml` ? Could you try renaming it back, or investigating why that check is not running anymore?

Everything else in the PR looks good to me!


<img width="353" alt="image" src="https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/assets/27730963/a2dd0462-162a-41e1-9f26-88ae4344359c">

harshala-rajesh commented:
> @harshala-rajesh it seems that the `[Yamato] Pull Request Trigger` part of the CI is not running, could it be because it expects a file called `tests.yml` , which was renamed in favor of `shared_triggers.yml` ? Could you try renaming it back, or investigating why that check is not running anymore?
> 
> Everything else in the PR looks good to me!
> 
> <img alt="image" width="353" src="https://private-user-images.githubusercontent.com/27730963/343758099-a2dd0462-162a-41e1-9f26-88ae4344359c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NDUyMzMsIm5iZiI6MTcxOTk0NDkzMywicGF0aCI6Ii8yNzczMDk2My8zNDM3NTgwOTktYTJkZDA0NjItMTYyYS00MWUxLTlmMjYtODhhZTQzNDQzNTljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE4Mjg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0NjYxZjNmY2NjOGNlMDhlZDg3Yzg4OGJlMzBkYzJkYTE2MWNiMGYzM2U5Y2U3OWU5Y2MzZjMzMjZiNzBmODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cyibOUKomMVgVlDOa-Doaks1N6c-IsrN4ZrOW-0MiVo">

@RikuTheFuffs - sorry , I should not have deleted tests.yml , added it back.
For the [Yamato] Pull Request Trigger (main, develop, & sample branches)  which is showing up as required - this is the trigger I split into multiple triggers,I deleted the original trigger before it finished the run on first push. Hence we see the message. Can it be merged in this state ?

RikuTheFuffs commented:
> I should not have deleted tests.yml , added it back. For the [Yamato] Pull Request Trigger (main, develop, & sample branches) which is showing up as required - this is the trigger I split into multiple triggers,I deleted the original trigger before it finished the run on first push. Hence we see the message. Can it be merged in this state ?

Let's try to rerun it first to see what happens. All checks must pass in order for the PR to be merged, otherwise it can cause a stuck CI in all future pull requests.

RikuTheFuffs commented:
> I should not have deleted tests.yml , added it back.

@harshala-rajesh about this, could you explain why not? Aren't these the "old" tests that triggered regardless of where the changes were made? Or do they have some additional logic that is not covered by the shared_tests?

RikuTheFuffs reviewed (approved):
pre-approving while we wait for an answer about shared_triggers vs triggers.yml
